library/Asana/Api/Generic.hs

```diff
@@ -0,0 +1,11 @@
+module Asana.Api.Generic (GenericAsanaTask) where
+
+import Data.Aeson (FromJSON)
+
+-- Class of things that can be parsed as "tasks"
+--
+-- Since Asana's API is extensible, it makes sense to allow the user some
+-- ability to change the way a @Task@ is parsed. This gives us a little bit of
+-- flexibility when it comes to the myriad ways a @Task@ list can be fetched
+-- (see @getProjectTasks@)
+class FromJSON a => GenericAsanaTask a
```

Comment:
This doesn't have any methods, so I'm not sure what it's adding. If it's for documentation in contexts, can we do the old:

```haskell
class FromJSON a => GenericAsanaTask a
instance FromJSON a => GenericAsanaTask a
```
?

Reply:
This is a way to tag a representation of an asana task.

The list above explains it a bit: api "embeds" can be added in asana by specifying a list of extra fields you want the response to supply. However, we don't always want to parse those fields. So it is left up to the user to _decide_ on a structure (or use the one we provide in the library).

This is used in the `planning-poker` executable, where we request the `projects` for a task when we fetch it, and ask for a task _with projects_.

It's not really for documentation purposes as much as trying to make the intent clearer: When you fetch a task, you must provide a data type to parse into. Instantiating `GenericAsanaTask` is to say "you can parse a _task_ into my thing," which is more explicit than saying "fetch a task, then parse it into this thing _that can be parsed as json_", which is all a `FromJSON` instance would imply.

I don't know if I explained that super well, let me know if you're still confused about the API design or think it's a dumb idea!

Reply:
Mm, I think I get it, I'm just not sure what it buys over `FromJSON` other than documentation. I don't typically use empty typeclasses unless I'm defining all instances (and trying to _hide_ superclasses) or to join multiple superclasses together. I don't feel strongly about it though.